### DISCUSSION TITLE
solved: Issue: Unable to wipe between the NSPanel screens, after sending MQTT messages regarding widgets

### DISCUSSION THREAD

**Discussion by issleiber** [blakadder/nspanel#3] [General]
First of all: Thank's for the good job,

I have one persistent problem. 

Nearly everything works fine. But if I send an MQTT message regarding the widgets (or use the tasmota console instead), after that, I cannot switch/wipe the Screens on my NSPanel. Sometimes, I can only switch one time (e.g. from the Widget View to the heater widget view). No matter what MQTT Messages (regarding widgets) I send.

**short issue description:**

1.) Start my NSPanel, MQTT Messages from NSPanel to my MQTT Broaker works fine. I receive everything, what I expect to receive
2.) I can switch all the views on my NSPanel (wiping the Screen) from the SwitchWidgets to Thermostat ... and so on 
3.) But! if I send a MQTT Message to my NSPanel (regarding the Widgets) .... e.g.

`cmnd/NSPanel/nspsend {"HMI_resources":[{"index":7,"ctype":"group","id":"Licht","uiid":6}]}`

or any other MQTT like ..

`cmnd/NSPanel/nspsend {"index":7,"type":"delete"}` ...

I cannot wipe the Screen on my NSPanel (or sometimes I can wipe only once to the (e.g.) thermostat and not wipe back or further). 

This behaviour takes place only if I send MQTT Message regarding the Widgets. Other MQTT messages, except Widgets,  like 'cmnd/NSPanel/POWER1 ON' doesn't trigger this behaviour.

add 1: This behaviour also exist, if I use the console commands in "tasmota" like ... `nspsend {"index":7,"type":"delete"}`

Thanks in advance 

Andreas

**Comment by blakadder** [comment]
As noted in the documentation I collected you cannot delete only one widget, you need to configure all 8 at the same time so they init correctly

**Reply by issleiber** [reply]
Thank you, Yes it works.

One more questions: Is there any way to change the name of the double, triple - buttons (current names: OUTLET1, OUTLET2 ...) widget ? I can only change the name of the switch.

**Reply by blakadder** [reply]
I don't know where you got those names. Default naming scheme is `Index 1`, `Index 2`, etc. Those you change in https://github.com/blakadder/nspanel/blob/ecedb54c8c0e3df32df98a84cc752c8d217f5267/nspanel.be#L25

**Reply by issleiber** [reply]
Yes, I know that name scheme, but how can I change the names of the button part .... see picture ...
<img width="311" alt="Bildschirmfoto 2021-12-24 um 11 51 21" src="https://user-images.githubusercontent.com/3843337/147346833-e3244e61-e30c-4782-a2d9-2b33b612fd56.png">

> 
The description/names "OUTLET1 ... 4" for example ...

**Reply by blakadder** [reply]
I wasn't aware you can change those in the original firmware

**Comment by issleiber** [comment]
Some further Widget on NSPanel found:

  6: ["Curtain", "device", 11] .... is a curtain/blinds Widget (0-100%) with submenues 

I'll try to figure out some more

**Reply by blakadder** [reply]
which firmware version is that on?

**Comment by issleiber** [comment]
I didn't know an easy way to figure out the firmware version of the nspanel. I use the latest tasmota (tasmota32) image and I also have the US version of the NSPanel (which (I guess) doesn't matter)

MQTT messages regarding curtain widget:

```
tele/NSPanel/RESULT = {"NSPanel":{"id":"5","params":{"setclose":27}}}... send, when the curtain is close with 27% 

tele/NSPanel/RESULT = {"NSPanel":{"id":"5","params":{"switch":"on"}}} .... send, when the curtain completely open 

tele/NSPanel/RESULT = {"NSPanel":{"id":"5","params":{"switch":"off"}}} .... send, when the curtain completely closed

tele/NSPanel/RESULT = {"NSPanel":{"id":"5","params":{"switch":"pause"}}} .... send, when the curtain stop/pause moving
```

![IMG_4957](https://user-images.githubusercontent.com/3843337/147768600-bfb9de78-e318-4fae-bcbe-0431c75b40c9.jpeg)
.
